pkg/apply/merge.go

```diff
@@ -112,6 +116,51 @@ func MergeServiceAccountForUpdate(current, updated *unstructured.Unstructured) e
 	return nil
 }
 
+// MergeWebhookConfiguration ensure caBundle is keept at webhooks's clientConfig
+func MergeWebhookConfiguration(current, updated *unstructured.Unstructured) error {
+	gvk := updated.GroupVersionKind()
+	if gvk.Kind == "MutatingWebhookConfiguration" || gvk.Kind == "ValidatingWebhookConfiguration" {
+		currentWebhooks, found, err := unstructured.NestedSlice(current.Object, "webhooks")
+		if err != nil {
+			return errors.Wrap(err, "failed searching for webhooks at current configuration")
+		}
+		if !found {
+			return nil
+		}
+
+		updatedWebhooks, found, err := unstructured.NestedSlice(updated.Object, "webhooks")
+		if err != nil {
+			return errors.Wrap(err, "failed searching for webhooks at updated configuration")
+		}
+		if !found {
+			return nil
+		}
+		for ci, _ := range currentWebhooks {
```

Comment:
Do we need to iterate ^2? Would you please create two dictionaries `currentWebhookByName` and `updatedWebhookByName` and then compare those with instant access?

Reply:
super early optimization, we are not going to have a massive number of of webhooks.

Reply:
It is not only to optimize the performance but also readability. Precreating of two maps is a clear and well understood operation. Iterating one of them and looking up keys from the other as well. Having two nested loops with the `if` in it is not that clear IMO.

Reply:
@qinqon I want to finish this argument